mlrun/model.py

```diff
@@ -546,6 +547,33 @@ def __init__(
         self.status = status
         self.sent_time = sent_time
 
+        self.validate_notification()
+
+    def validate_notification(self):
+        if self.kind:
+            try:
+                mlrun.common.schemas.notification.NotificationKind(self.kind)
+            except ValueError as exc:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    f"notification kind {self.kind} is not supported"
+                ) from exc
+
+        if self.severity:
+            try:
+                mlrun.common.schemas.notification.NotificationSeverity(self.severity)
+            except ValueError as exc:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    f"notification severity {self.severity} is not supported"
+                ) from exc
+
+        if self.status:
+            try:
+                mlrun.common.schemas.notification.NotificationStatus(self.status)
+            except ValueError as exc:
+                raise mlrun.errors.MLRunInvalidArgumentError(
+                    f"notification status {self.status} is not supported"
+                ) from exc
+
```

Comment:
wouldn't it be easier if we will have a schema for the notification struct and then validate it against it, that way the validation code would be more concise and would automatically add new attributes ones they added